### ISSUE TITLE
Backup not working

### ISSUE THREAD

**Issue by RadioactiveRadio** [darkmoonight/ToDark#18]
I tried to backup my to-do's but it keeps saying "Something went wrong" no matter where i try to save it. I was doing it on the sd card at first and thought that was the problem but it was doing that for the internal storage as well.

Same behavior when clicking on restore.

**Comment by Leonavichus** [comment]
Where did you download the app from?

**Comment by RadioactiveRadio** [comment]
From f-droid

**Comment by Leonavichus** [comment]
https://apt.izzysoft.de/fdroid/index/apk/com.yoshi.todark ?

**Comment by RadioactiveRadio** [comment]
Yes

**Comment by Leonavichus** [comment]
Then it's weird.

**Comment by Leonavichus** [comment]
Can you tell me what kind of device you have and what version of Android?

**Comment by RadioactiveRadio** [comment]
It's Android 8, Samsung A9 pro (2016 model)

**Comment by mojienjoyment** [comment]
I have this issue, too. Downloaded from GitHub releass. when I restore to-do, It says " Something went wrong"

**Comment by Leonavichus** [comment]
If this is still relevant for you, can you see if everything works?

**Comment by mojienjoyment** [comment]
@Leonavichus
Tested on 1.4.0 and I can't create backup! It shows me "sth went wrong". In some previous versions, I created backup but couldn't reatore. In this version I can't create backup file...

**Comment by mojienjoyment** [comment]
> If this is still relevant for you, can you see if everything works?

I test it again and find out that I just can create backup on certain folders! I created some folders on root or other subdirectories but I just can use"Documents" folder and their subfolders and "Downloads" subfolders!

another things that I can restore them but incomplete! for example when restore "task.json", just categories restored and after that when I restore "todo.json", It shows me "sth went wrong" but restore just one of my tasks in "Restored tasks" categories, not my categories. Is it confusing?

**Comment by Leonavichus** [comment]
> > If this is still relevant for you, can you see if everything works?
> 
> I test it again and find out that I just can create backup on certain folders! I created some folders on root or other subdirectories but I just can use"Documents" folder and their subfolders and "Downloads" subfolders!
> 
> another things, I can restore them but not completely! for example when restore "task.json", just categories restored and after that when I restore "todo.json", It shows me "sth went wrong" but restore just one of my tasks in "Restored tasks" categories, not my categories. Is it confusing?

Yes, I also just discovered this error, I will try to fix it this week.

**Comment by Leonavichus** [comment]
@mojienjoyment, You can see if everything is working fine? As for the fact that the data is not being restored in your categories, I can't do anything about it yet, since isar (the database I use) will not be able to export links.

**Comment by mojienjoyment** [comment]
> @mojienjoyment, You can see if everything is working fine? As for the fact that the data is not being restored in your categories, I can't do anything about it yet, since isar (the database I use) will not be able to export links.

Everything are good. all my task restored! All my categories are restored (blank). previous problems solved. backup folder problem still there. I can just choose some folders for backup folder location

**Comment by RadioactiveRadio** [comment]
No, it's not working as far as i can tell. i tried to save it but it game me the same "Something went wrong." message.

**Comment by OsitoGrande** [comment]
I have the same issue. Android 14, November 2023 security update, v1.46 of the app from GitHub. "Something went wrong" after trying to allow giving it access to a specific backup directory.
Does the app need any additional storage permissions assigned to it?

**Comment by f1refa11** [comment]
checked app logs:

![Screenshot_20240102-212345_MatLog~2.png](https://github.com/darkmoonight/ToDark/assets/58619409/87e49eba-eba5-4509-9468-8246b74d4171)

seems that the OS just somehow can't give folder access to the app

**Comment by Leonavichus** [comment]
I'll try to fix it.
